File: src/black_russia.py
from requests import get

class BlackRussia:
	def __init__(self) -> None:
		self.api = "http://api.blackrussia.online"
		self.headers = {
			"user-agent": "Dalvik/2.1.0 (Linux; U; Android 9; RMX3551 Build/PQ3A.190705.003)"
		}

	def get_donate_items(self) -> dict:
		return get(
			f"{self.api}/donate-items.json",
			headers=self.headers).json()

	def get_admin_list(self) -> list:
		return get(
			f"{self.api}/admins-list.json",
			headers=self.headers).json()

	def get_car_colors(self) -> dict:
		return get(
			f"{self.api}/car-colors.json",
			headers=self.headers).json()

	def get_servers(self) -> list:
		return get(
			f"{self.api}/servers.json",
			headers=self.headers).json()

	def get_black_pass(self) -> dict:
		return get(
			f"{self.api}/black-pass.json",
			headers=self.headers).json()

	def get_tune_items(self) -> dict:
		return get(
			f"{self.api}/tune-items.json",
			headers=self.headers).json()

	def get_tune_screens(self) -> list:
		return get(
			f"{self.api}/tune-screens.json",
			headers=self.headers).json()

	def get_tune_vinyls(self) -> list:
		return get(
			f"{self.api}/tune-vinyls.json",
			headers=self.headers).json()

	def get_cars_and_accessories(self) -> dict:
		return get(
			f"{self.api}/cars-and-accessories.json",
			headers=self.headers).json()

	def get_inventory(self) -> dict:
		return get(
			f"{self.api}/client/inventory.json",
			headers=self.headers).json()

	def get_social_interaction(self) -> dict:
		return get(
			f"{self.api}/client/social-interaction.json",
			headers=self.headers).json()

	def get_family_system(self) -> dict:
		return get(
			f"{self.api}/client/family_system.json",
			headers=self.headers).json()

	def get_stories(self) -> list:
		return get(
			f"{self.api}/stories.json",
			headers=self.headers).json()

File: src/black_russia.js
class BlackRussia {
	constructor() {
		this.api = "http://api.blackrussia.online"
		this.headers = {
			"user-agent": "Dalvik/2.1.0 (Linux; U; Android 9; RMX3551 Build/PQ3A.190705.003)"
		}
	}

	async getDonateItems() {
		const response = await fetch(
			`${this.api}/donate-items.json`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getAdminList() {
		const response = await fetch(
			`${this.api}/admins-list.json`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getCarColors() {
		const response = await fetch(
			`${this.api}/car-colors.json`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getServers() {
		const response = await fetch(
			`${this.api}/servers.json`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getBlackPass() {
		const response = await fetch(
			`${this.api}/black-pass.json`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getTuneItems() {
		const response = await fetch(
			`${this.api}/tune-items.json`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getTuneScreens() {
		const response = await fetch(
			`${this.api}/tune-screens.json`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getTuneVinyls() {
		const response = await fetch(
			`${this.api}/tune-vinyls.json`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getCarsAndAccessories() {
		const response = await fetch(
			`${this.api}/cars-and-accessories.json`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getInventory() {
		const response = await fetch(
			`${this.api}/client/inventory.json`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getSocialInteraction() {
		const response = await fetch(
			`${this.api}/client/social-interaction.json`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getFamilySystem() {
		const response = await fetch(
			`${this.api}/client/family_system.json`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getStories() {
		const response = await fetch(
			`${this.api}/stories.json`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}
}

module.exports = {BlackRussia}
